### PR TITLE
docs: FIPS support

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Officially claim FIPS 140-3 support. Add a document explaining how the binaries can be obtained, and how to verify their correctness.

Delete remaining references to FIPS builds on darwin. BoringCrypto is only certified for Linux on amd64.

### CODE REVIEW

File: docs/fips.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+While [FIPS 140-3][fips] doesn't, strictly speaking, mandate a specific TLS version, it is nonetheless a good practice to use as recent of a version as possible. Otelcol allows TLS to be configured for each component individually, in a standardized manner. You can find the details [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md).
+
+For example, here is how we can set a minimum TLS version of 1.2 in the `otlp` exporter:
+
+```yaml
+exporters:
+  otlp:
+    endpoint: myserver.local:55690
+    tls:
+      min_version: "1.2"
```
Comment (perk) [inline docs/fips.md:191]: Isn't that the default actually?
If yes I think we should call it out.

Right now, the way it's written one might understand that they need to set it explicitly to have TLS 1.2

**Review by perk — APPROVED** [review]
LGTM except for one comment.

File: docs/fips.md (hunk 1 above)
Comment (swiatekm-sumo) [reply docs/fips.md:191]: It is the default, yeah. I'll call it out as you suggest.

Comment (swiatekm-sumo) [reply docs/fips.md]: Done.

**Review by perk-sumo — APPROVED** [review]
LGTM 🚀

File: docs/fips.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
+# BoringCrypto and FIPS 140-3 compliance
```
Comment (dochel) [inline docs/fips.md:1]: Where did you get FIPS 140-3 from? The last module validation of Boring I see is still against FIPS 140-2? (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4407)

Maybe just say "FIPS compliance" and omit the 140-3?

File: docs/fips.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+## Introduction
+
+Sumo OT distro can be built to use BoringCrypto instead of Go standard library crypto implementations.
```
Comment (dochel) [inline docs/fips.md:5]: Maybe add something like "BoringCrypto is a cryptographic module validated against FIPS 140."

(Would it help to add a link to https://csrc.nist.gov/projects/cryptographic-module-validation-program/validated-modules/search?SearchMode=Basic&ModuleName=boring&CertificateStatus=Active&ValidationYear=0?)

File: docs/fips.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+```text
+docker pull public.ecr.aws/sumologic/sumologic-otel-collector:0.69.0-sumo-0-fips
+```
+
+> **Note**
+> Currently FIPS-compliant builds are only available for Linux on amd64.
+
+## Running in FIPS-approved mode
+
+The FIPS-compliant build statically enforces use of [FIPS 140-3][fips] compliant TLS settings, and doesn't require any
```
Comment (dochel) [inline docs/fips.md:48]: This is confusing, as it conflates two different things: FIPS 140 validation of the BoringCrypto module, and settings that ensure that TLS 1.2 is used only with cipher suites that use approved primitives. (FIPS 140 does not regulate TLS.)

Maybe say something like "The FIPS-compliant build statically enforces the use of FIPS-approved algorithms in TLS cipher suites, and..." ?

File: docs/fips.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+client curve: secp384r1 (24)
+client curve: secp521r1 (25)
+client point: uncompressed (0)
+
+[...]
+```
+
+## Setting minimum TLS version
+
+While [FIPS 140-3][fips] doesn't, strictly speaking, mandate a specific TLS version, it is nonetheless a good practice to use as recent of a version as possible. Otelcol allows TLS to be configured for each component individually, in a standardized manner. You can find the details [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md).
```
Comment (dochel) [inline docs/fips.md:182]: Specifically, versions earlier than 1.2 are considered unsafe. You might find differing opinions in the security/compliance community over whether 1.2 or 1.3 is the best option, currently.

File: docs/fips.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+By default, the minimum TLS version for otel components is `1.2`. If you want to set a higher one, here's an example of using `1.3`
+for the `otlp` exporter:
```
Comment (dochel) [inline docs/fips.md:185]: Interesting discussion here that hints that 1.3 might use non-approved algorithms? I know nothing about it, just happened to come across this write-up: https://kupczynski.info/posts/fips-golang/

Doesn't invalidate what you're saying, though. :)

**Review by dochel — COMMENTED** [review]
Added some comments...

File: docs/fips.md (hunk 2 above)
Comment (swiatekm-sumo) [reply docs/fips.md:1]: I swear I saw a cert for 140-3, but it's clearly not actually there. Sorry for the confusion.

File: docs/fips.md (hunk 6 above)
Comment (swiatekm-sumo) [reply docs/fips.md]: I specifically noted that `1.2` is safe to use, and that the status of `1.3` is uncertain as of right now.

**Review by dochel — APPROVED** [review]
Looks good to me. Thank you! :)

**Pull request merged**